course-content/31_exploring_data_structures.qmd

```diff
@@ -0,0 +1,295 @@
+
+```{python}
+df.head(2)
+```
+
+The `.head()` method let's us look at actual data like we would likely do in Excel. This dataset has some high level country metrics about population and the economy going back to 1952. 
+
+### Selecting columns (variables)
+
+In the "real world" of data its not uncommon to see hundreds or thousands of columns, so knowing how to pare down our dataset so it is not overly bloated is important. Our dataset is just 6 columns wide, so not too big, but let's go ahead and narrow it down anyway.  
+
+Let's create an object (in this case a dataframe) called "narrow_df" and then specify the names of columns that we want to have in our slimmed down dataset. We are primarily interested in population, but we want to have year, country, and continent in there too. After we create it let's take a look at the new dataframe:
+
+```{python}
+narrow_df = df[['country', 'year', 'pop', 'continent']]
+narrow_df.head(2)
+```
+
+So now we have a more manageable dataset to work with called narrow_df. The original dataframe df is still there, and still the same size as before it we want to have a look though. 
+
+:::{.callout-tip}
+## Reminder on Assignment!
+When you create a variable or dataframe in python, to the left of the `=` sign you always put the name of the thing you want to make or modify. To the right, that's where you put the contents of what it is you want to create. It may feel counterintuitive! Especially if you are using the same variable or dataframe identifier on both sides of the equals sign.
+:::
+
+:::{.callout-warning icon="false"}
+
+## Challenge 1
+
+
+
+Let's say you just want to narrow down the dataset to include just the country and the GDP per capita. How would you do it? Don't forget to run your code so it also shows a view of the result so you can confirm the code worked as you wanted it to. 
+
+:::
+
+
+
+:::{.callout-note icon="false" collapse="true"}
+
+### Solution to Challenge 1
+
+First you would like to create a new object to hold this narrowed down version of the dataset, then just use some method like `.head()` to see some of the resulting data. 
+
+```{python}
+challenge1_df = df[['country', 'gdpPercap']]
+challenge1_df.head(2)
+```
+
+:::
+
+### Selecting Rows
+
+Of course, in data analysis we are usually interested in looking at just some of rows of data, not all of it all the time. So when we want to look at just selected rows (i.e. values within a given column) there is a handy pandas function called `.query()` that we can leverage. It helps us with selecting the rows we want. 
+
+We set up an expression in pandas  with `.query()` and it will evaluate it for us. For example, imagine we want to find just the rows where the year is 1972. Here's how we do that, using `==` instead of just `=`:
```

Comment:
The boolean `==` vs `=` for assignment can really trip people up... will probably need to spend a bit of time on this

Reply:
Good idea. Yes, I will explain the subtle difference and why/how it matters with code